src/app/routes/utils/constructPageFetchUrl/index.ts

```diff
@@ -0,0 +1,94 @@
+import Url from 'url-parse';
+import pipe from 'ramda/src/pipe';
+import getEnvironment from '#app/routes/utils/getEnvironment';
+import { getUrlPath } from '../../../lib/utilities/urlParser';
+import handleError from '../handleError';
+import { Services, Variants } from '../../../models/types/global';
+
+interface UrlConstructParams {
+  pathname: string;
+  pageType: 'article' | 'cpsAsset' | 'topic' | 'live';
+  service: Services;
+  variant?: Variants;
+  page?: string;
+}
+
+const removeAmp = (path: string) => path.split('.')[0];
+const getArticleId = (path: string) => path.match(/(c[a-zA-Z0-9]{10}o)/)?.[1];
+const getTipoId = (path: string) => path.match(/(c[a-zA-Z0-9]{10}t)/)?.[1];
+const getCpsId = (path: string) => path;
+
+const getId = (pageType: string) => {
+  let getIdFunction;
+  switch (pageType) {
+    case 'cpsAsset':
+      getIdFunction = getCpsId;
+      break;
+    case 'topic':
+    case 'live':
+      getIdFunction = getTipoId;
+      break;
+    default:
+      getIdFunction = getArticleId;
+      break;
+  }
+  return pipe(getUrlPath, removeAmp, getIdFunction);
+};
+
+const constructPageFetchUrl = ({
+  pathname,
+  pageType = 'topic',
+  service,
+  variant,
+  page,
+}: UrlConstructParams) => {
+  const env = getEnvironment(pathname);
+  const isLocal = !env || env === 'local';
+
+  const id = getId(pageType)(pathname);
+  const capitlisedPageType =
```

Comment:
typo
```suggestion
  const capitalisedPageType =
```

Reply:
Aside: Whyyyyyyyy is there no `capitalise` function on a string 😭

Reply:
<img src="https://media4.giphy.com/media/gd09Y2Ptu7gsiPVUrv/giphy.gif"/>